code_generation/quantity.py

```diff
@@ -33,20 +34,30 @@ def get_leafs_of_scope(self, scope):
         return [self.name] + [self.name + shift for shift in self.get_shifts(scope)]
 
     def shift(self, name, scope):
+        if scope in self.ignored_shifts.keys():
+            if name in self.ignored_shifts[scope]:
+                log.debug("Ignoring shift {} for quantity {}".format(name, self.name))
+                return
```

Comment:
This is a bit inconsistent, in this case the function returns a None, and in all other cases nothing is returned. Either all possible cases should return something, or no case should return something. So i would suggest to remove the return and use an else for the other cases

Reply:
I think a python function without return statement returns None as well.